pylinalg/func/vector.py

```diff
@@ -116,3 +116,195 @@ def vector_apply_matrix(vectors, matrix, /, *, w=1, out=None, dtype=None):
     if dtype is not None:
         out = out.astype(dtype, copy=False)
     return out
+
+
+def vector_unproject(vector, projection_matrix, /, *, depth=0, out=None, dtype=None):
```

Comment:
this function should broadcast over many vectors if possible, like the other functions in `vector.py` do

Reply:
@Korijn Sure. Should that work with only a single `matrix` (that is broadcasted to match the number of vectors) or with a batch of matrices, too.